### PR TITLE
Support cluster lifecycle operations on v1alpha2 based TKGS clusters

### PR DESCRIPTION
**Pull request by prkalle**
Signed-off-by: PremKumar Kalle <pkalle@vmware.com>

**What this PR does / why we need it**:
This PR would support cluster lifecycle operations on v1alpha2 base TKGS clusters.
User can create a TKC cluster with one nodepool. However user can perform CRUD operation on nodepools using `tanzu cluster nodepool set/delete/list` command. 
**Which issue(s) this PR fixes**:

Fixes #

**Describe testing done for PR**:
Created Tanzu kubernetes cluster with the v1alpha2 API version and was success
Deleted Tanzu kubernetes cluster with the v1alpha2 API version and was success
Upgraded Tanzu kubernetes cluster with the v1alpha2 API version and was success
`tanzu cluster list` command was successfully showing the v1alpha2 API version TKC clusters

**Special notes for your reviewer**:
1) Support for nodepool CRUD operation for TKGS provider with v1alpha2 API version would be added in a separate PR.
2) Suppprt for Scaling operation for TKGS provider with v1alpha2 API version would be added in a separate PR. 
**Release note**:

```release-note
Added support for cluster lifecycle operations on v1alpha2 based TKGS clusters
```
**New PR Checklist**

- [x] Ensure PR contains only public links or terms
- [x] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [x] Squash the commits in this branch before merge to preserve our git history
- [x] If this PR is just an idea or POC, use a [Draft PR](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/about-pull-requests#draft-pull-requests) instead of a full PR
- [x] Add appropriate [kind label](../docs/release/kind-labels.md) according to what type of issue is being addressed.

### CODE REVIEW

File: apis/run/v1alpha2/groupversion_info.go [hunk 1]
```diff
@@ -0,0 +1,23 @@
+// Copyright (c) 2021 VMware, Inc. All Rights Reserved.
```
Comment (saimanoj01) [inline apis/run/v1alpha2/groupversion_info.go:1]: Q: Did TKGS already release a version supporting v1alpha2? I am asking to know if the API types are finalized or if there is any scope for change.

File: pkg/v1/tkg/clusterclient/clusterclient.go [hunk 2]
```diff
@@ -1578,8 +1602,8 @@ func (c *client) ScalePacificClusterControlPlane(clusterName, namespace, apiVers
 	return nil
 }
 
-func (c *client) ScalePacificClusterWorkerNodes(clusterName, namespace, apiVersion string, workersCount int32) error {
-	tkcObj, err := c.getPacificClusterObject(clusterName, namespace, apiVersion)
+func (c *client) ScalePacificClusterWorkerNodes(clusterName, namespace string, workersCount int32) error {
+	tkcObj, err := c.getPacificClusterObject(clusterName, namespace)
```
Comment (saimanoj01) [inline pkg/v1/tkg/clusterclient/clusterclient.go:1606]: Is the plan to revisit the scale logic with a different PR?

**Comment by saimanoj01** [comment]
> Cluster Generation A/B Results: https://storage.googleapis.com/tkg-clustergen/759/20211004172707/clustergen.diff.txt Author/reviewers: Please review to verify that the effects on the generated cluster configurations are exactly what the PR intended, and give a thumbs-up if so.

I don't see any diffs in the clustergen tests which reflects the template changes. Did you get a chance to look into it?

File: pkg/v1/providers/infrastructure-tkg-service-vsphere/v1.1.0/ytt/overlay.yaml [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    controlPlane:
+      vmClass: #@ data.values.CONTROL_PLANE_VM_CLASS
+      replicas: #@ data.values.CONTROL_PLANE_MACHINE_COUNT
+      storageClass: #@ data.values.CONTROL_PLANE_STORAGE_CLASS
+      tkr:
+        reference:
+          name: #@ data.values.KUBERNETES_RELEASE
+    nodePools:
+    #@overlay/match by=overlay.index(0)
+    - name: #@ data.values.NODE_POOL_0_NAME
```
[the remark this answers is not in the document]
Comment (prkalle) [reply pkg/v1/providers/infrastructure-tkg-service-vsphere/v1.1.0/ytt/overlay.yaml:73]: I made the `NODE_POOL_0_NAME` as required configuration parameter and CLI would throw error if this parameter is not set during cluster creation. But I am open to ssuggestion making a default value for `NODE_POOL_0_NAME` to `"workers"`
cc: @vuil

File: pkg/v1/tkg/clusterclient/clusterclient.go (hunk 2 above)
Comment (prkalle) [reply pkg/v1/tkg/clusterclient/clusterclient.go:1606]: Yes. The plan is to have seperate PR for scaling logic for TKGS

File: pkg/v1/providers/infrastructure-tkg-service-vsphere/v1.1.0/ytt/overlay.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (prkalle) [reply pkg/v1/providers/infrastructure-tkg-service-vsphere/v1.1.0/ytt/overlay.yaml:73]: updated with default value for `NODE_POOL_0_NAME` to `workers`

**Review by anujc25 — APPROVED** [review]
lgtm.

**Pull request merged**